frontend/src/contexts/AuthContext.tsx

```diff
@@ -35,7 +41,15 @@ export const useAuth = (): AuthContextProps => {
 
 // Provider hook that creates auth object and handles state
 const useProvideAuth = () => {
-  const [isAuthenticated] = useLocalStorage<boolean>(LOGGED_IN_KEY)
+  // TODO #4279: Remove after React rollout is complete
+  const { data: user } = useQuery<UserDto>(
+    userKeys.base,
+    () => fetchUser(),
+    // 10 minutes staletime, do not need to retrieve so often.
+    { staleTime: 600000 },
```

Comment:
should we still add an `onSuccess` to set localStorage to the authenticated state?

Reply:
oh yes good point! added